### ISSUE TITLE
No examples or  docs about non "EZ" interfaces to C++ RPC.

### ISSUE THREAD

**Issue by adrianratnapala** [sandstorm-io/capnproto#357]
### Observation

The existing documentation only shows how to do C++ RPC using `ez-rpc`.  But thouse same docs point out that `ez-rpc` is only a limited set of use-cases (as explained in  `ez-rpc.h`).  

Everything in the `c++/samples/` directory either uses `ez-rpc` or does no RPC at all.  Also the sources of `ez-rpc` itself don't seem designed as a simple-to-read example.

### Expectation

There should be some documentation or example for how to use the lower-level interfaces.

### Example

My particular problem is that I am trying to do RPC over a socketpair.  I think I can use a `EzRpcClient` for this.  But `EzRpcServer` seems to expect a socket that it can `listen()` on, rather than an already-connected socket.

From reading the source I *think* I can solve this problem by incantation-involving a `kj::UnixEventPort`, a `kj::UnixEventPort::FdObserver` and a `kj::EventLoop`.  But some guidance from would be nice.

Cheers.

**Comment by kentonv** [comment]
Yeah, the trouble is, there is a lot to document. The header files usually provide pretty good reference docs in the comments, FWIW.

The basic way to set up RPC without `EzRpc` is:

1. Use `kj::setupAsyncIo()` to set up the event loop. It returns a bundle of interfaces that let you do a variety of things low-level (wrapping FDs) and high-level (connecting to hosts by name), producing `kj::AsyncIoStream`s and `kj::ConnectionReceiver`s.
2. Construct `capnp::TwoPartyClient` or `capnp::TwoPartyServer` to set up RPC. These let you pass in arbitrary `kj::AsyncIoStream`s and `kj::ConnectionReceiver`s.

**Comment by adrianratnapala** [comment]
Thanks.   I look forward to having time to try it out.  If I do, I will try to document my experiences and contribute the those docs.  No guarantees though.

**Comment by rustyx** [comment]
Would it be possible to add a simple calculator example using `kj` methods directly instead of using `EzRpc`? It's a pain to have to reverse-engineer `EzRpc`, just to be able to understand how RPC is supposed to be used.

Also would be really nice to have an example with TLS. I don't even know if TLS is supported. I found `TlsContext`, but it's not documented anywhere.
